wagtail/tests/test_page_model.py

```diff
@@ -3917,3 +3931,20 @@ def test_changes_when_slug_changes(self):
         self.page.slug = "something-else"
         self.page.save()
         self.assertNotEqual(self.page.cache_key, original_cache_key)
+
+
+class TestPageCachedParentObjExists(TestCase):
+    fixtures = ["test.json"]
+
+    def test_cached_parent_obj_exists(self):
```

Comment:
I wonder if this one could be optimised somewhat, for example: just take the root page, set `_cached_parent_obj` to some nonsense variable, then check `get_parent()` returns the nonsense variable. The assertion failure message could even say something super obvious, like:
> Page.get_parent() no longer uses _cached_parent_obj as a cache. We  should probably revise how this is used  to optimise page routing

What do you think?